tools/sigclient/pkg/utils/calculation.go

```diff
@@ -110,3 +110,60 @@ func ConvertStringToEpochSec(nowTs uint64, inp string, defValue uint64) uint64 {
 	}
 	return retVal
 }
+
+func RemoveValues[K comparable, T any](values []K, valuesToRemove map[K]T) []K {
+	finalValues := []K{}
+	for _, value := range values {
+		if _, exist := valuesToRemove[value]; !exist {
+			finalValues = append(finalValues, value)
+		}
+	}
+
+	return finalValues
+}
+
+func CompareSlices[K comparable](a []K, b []K) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i := range a {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
+func ElementsMatch[K comparable](a []K, b []K) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	mp := make(map[K]int)
+	for _, v := range a {
+		mp[v]++
+	}
+	for _, v := range b {
+		if _, ok := mp[v]; !ok {
+			return false
+		}
+		mp[v]--
+		if mp[v] == 0 {
+			delete(mp, v)
+		}
+	}
+
+	return len(mp) == 0
+}
+
+func AlmostEqual(a, b, tolerance float64) bool {
+	if a == b {
+		return true
+	}
+	diff := math.Abs(a - b)
+	if b == 0 {
+		return diff < tolerance
+	}
+
+	return (diff / math.Abs(b)) < tolerance
+}
```

Comment:
I'm a little confused why this function isn't simply `Abs(a - b) <= tolerance`. Is `tolerance` supposed to be an error percentage rather than an absolute error? If so, you should specify that in the name, and also in that case I think (?) it matters whether `a` is the `actual` and `b` is the `expected or if it's the other way around; if it does matter, you should rename them

Reply:
addressed in the latest commit.